### PR TITLE
test: Add component test for the adapter service

### PR DESCRIPTION
**Pull request by danielpacak**
This one allows me to run a out of process component test in semi-automated way. By component test I mean the aquasec/harbor-scanner-trivy container and its dependent services run with Docker Compose. The component test itself acts as a Harbor client testing the happy path with a sample image reference which is pulled from Docker Hub, retagged and then pushed to the test registry. All the corner cases and error paths should be test with unit tests.

```
$ docker-compose up -d -f test/component/docker-compose.yaml
$ go test -v -run "^TestComponent$" -tags=component ./test/component/...
$ docker-compose down -f test/component/docker-compose.yaml
```

The next step would be to fully automate test execution with a single command `make test-component` but it requires additional work & config. I'll handle it in a separate PR according to #38 .

Here's the output form the test run:

component-test-not-fully-automated-but-super-cool

### CODE REVIEW

File: test/component/client.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+		return report, fmt.Errorf("invalid response status: %s", res.Status)
+	}
+
+	b, _ := ioutil.ReadAll(res.Body)
+	_ = res.Body.Close()
+
+	err = json.Unmarshal(b, &report)
+	if err != nil {
+		return
+	}
```
Comment (knqyf263) [inline test/component/client.go]: ditto

File: test/component/component_test.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	if err != nil {
+		return "", err
+	}
+	return registryURL.Host, nil
+}
+
+// TestComponent is a component test for the whole adapter service.
+// It should test only the shortest and happiest path to make sure that all pieces are nicely put together.
+func TestComponent(t *testing.T) {
+	if testing.Short() {
```
Comment (knqyf263) [inline test/component/component_test.go]: I didn't know it. Cool!

File: test/component/client.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
+package component
```
Comment (knqyf263) [inline test/component/client.go]: nits: If this file is used only for testing, it may be better to change the file name to `client_test.go`.
The binary may contain this file by mistake. 

>For example, if we would have a file called person.go its test file must be named person_test.go. This is due to the package building process, where Go knows to ignore these files when building the package due to their naming. 

https://ieftimov.com/post/testing-in-go-naming-conventions/

But I couldn't find this explanation in the official page. I might be too worried. If so, never mind.

File: test/component/client.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+		return scanResp, fmt.Errorf("invalid response status: %v %v", resp.StatusCode, resp.Status)
+	}
+
+	rb, _ := ioutil.ReadAll(resp.Body)
+	_ = resp.Body.Close()
+
+	err = json.Unmarshal(rb, &scanResp)
+	if err != nil {
+		return
+	}
```
[the remark this answers is not in the document]
Comment (danielpacak) [reply test/component/client.go]: Thanks for pointing this out. Really nice shortcut. Done!

File: test/component/client.go (hunk 3 above)
Comment (danielpacak) [reply test/component/client.go]: You're right @knqyf263 This should be called `client_test.go` for the reason you've mentioned. But also to not decrease our test coverage, i.e. codecov would complain that client.go does not have any tests ;)
